tools/accuracy_checker/accuracy_checker/metrics/reid.py

```diff
@@ -31,6 +29,32 @@
 PairDesc = namedtuple('PairDesc', 'image1 image2 same')
 
 
+def _average_binary_score(binary_metric, y_true, y_score):
+    def binary_target(y):
+        return not (len(np.unique(y)) > 2) or (y.ndim >= 2 and len(y[0]) > 1)
+
+    if binary_target(y_true):
+        return binary_metric(y_true, y_score, sample_weight=None)
+
+    y_true = y_true.ravel()
+    y_score = y_score.ravel()
+
+    if y_true.ndim == 1:
+        y_true = y_true.reshape((-1, 1))
+
+    if y_score.ndim == 1:
```

Comment:
Are all these conditions necessary? I would expect the number of dimensions to be known in advance.

Reply:
no, I checked, we no need that, thanks